DOMA-4058 fixed history changes after auto reopen ticket

Alllex202 opened the pull request:
Before:
![Снимок экрана 2022-08-31 в 17 57 20](https://user-images.githubusercontent.com/56914444/187683682-d9b768b5-78ab-432e-a559-c60b3e3e0e41.png)

After:
![Снимок экрана 2022-08-31 в 17 58 03](https://user-images.githubusercontent.com/56914444/187683789-e6d1a3ed-93b4-4963-8a29-c272b3467e9b.png)

Alllex202 commented on apps/condo/domains/ticket/components/TicketChanges/TicketChange.tsx:
```diff
@@ -316,11 +317,88 @@ const useChangedFieldMessagesOf = (ticketChange) => {
         ticketChange[`${field}From`] !== null || ticketChange[`${field}To`] !== null
     ))
 
+    if (ticketChange.sender.fingerprint === 'auto-reopen') {
```
ok, on line 264 it is also used.
this is the most efficient and simple method.
need to do additional checks for changes in the expected fields?

pahaz replied:
I don't have enough business context to advise. But, you need to find a better way. This field is not for business logic

Alllex202 commented:
Also:
 - refactored logic without use sender field. 
 - fixed color status in auto-close ticket change message
     - Before:
![Снимок экрана 2022-09-01 в 17 20 20](https://user-images.githubusercontent.com/56914444/187912830-d5c4130a-a8d0-4768-b9fe-b119f7a8952a.png)
     - After: 
![Снимок экрана 2022-09-01 в 17 20 03](https://user-images.githubusercontent.com/56914444/187912876-3c232cd6-a6cb-48e1-8e39-d7327c4bd137.png)

dkoviazin commented on apps/condo/domains/ticket/components/TicketChanges/TicketChange.tsx:
```diff
@@ -316,11 +333,88 @@ const useChangedFieldMessagesOf = (ticketChange) => {
+                    }}
+                />
+            )
+        } else if (field === 'statusDisplayName') {
```
Too many else if for one field conditions
You can use object with keys for field value 
Or switch ... case

pahaz commented on apps/condo/domains/ticket/components/TicketChanges/TicketChange.tsx:
```diff
@@ -316,11 +333,88 @@ const useChangedFieldMessagesOf = (ticketChange) => {
     }
+
+    const formatAutoReopenDiffMessage = (field, ticketChange) => {
+        if (field === 'deferredUntil') {
```
Use object/dict like elseIf pattern

pahaz commented on apps/condo/domains/ticket/components/TicketChanges/TicketChange.tsx:
```diff
@@ -260,17 +270,24 @@ const useChangedFieldMessagesOf = (ticketChange) => {
 
+        if (isAutoCloseTicketChanges(ticketChange)) {
+            return (
+                <FormattedMessage
```
we use another pattern to formatted messages. Don't use FormattedMessage tag

Alllex202 replied:
refactored it
